linux/LinuxProcessTable.c

```diff
@@ -1098,6 +1098,33 @@ static void LinuxProcessTable_readCwd(LinuxProcess* process, openat_arg_t procFd
    free_and_xStrdup(&process->super.procCwd, pathBuffer);
 }
 
+/*
+ * Read /proc/<pid>/syscall (process-shared data)
```

Comment:
I don't think this is process shared data, AFAIC a thread does not stall all threads, see https://lwn.net/Articles/799557/